### PR TITLE
MRG: Additions to fine-calc

### PR DESCRIPTION
**Pull request by wronk**
- [x] Indexing error
- [x] write fine calibration to `.dat` file
- [x] write round trip tests for loading/saving `.dat` files
- [ ] After merging, open PR to update `_read_fine_cal` function within `maxwell.py` to use the new version of the function in `_fine_calc.py` that passes the calib. info in a `dict`

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Want to fix the docstring while you're in there?

**Comment by wronk** [comment]
Yep, np

**Comment by wronk** [comment]
So there's a little bit of inconsistency in the current read/write functions.
`read_fine_cal` takes a filename and returns a tuple of the channel names and their fine cal info:
> return cal_chs, cal_ch_numbers

`write_fine_cal`, on the other hand, takes a `calibration` dict that currently must contain `ch_names`, `locs`, and `imb_cals` as keys. Should we standardize this? Do you care which function we remold? We usually want to get all the channel into one big array asap, so combining `locs` and `imb_cals` might make more sense.

**Comment by wronk** [comment]
Also, okay to put this load/write round trip test in test_maxwell.test_fine_calibration()

**Comment by Eric89GXL** [comment]
> read_fine_cal takes a filename and returns a tuple of the channel names and their fine cal info:

I would probably make both deal with the "complete representation", i.e. the dict. Then whatever uses `read_fine_cal` should take the dict convert to `cal_chs` and `cal_ch_numbers`. Make sense?

> Also, okay to put this load/write round trip test in test_maxwell.test_fine_calibration()?

It's a bit misleading because it's actually independent of Maxwell filtering. In principle fine cal could be used for other things, e.g. inversion. So I'd make a new file like `test_fine_cal.py` just for the IO. That way `test_maxwell` can be dedicated to actually doing MF.

File: mne/preprocessing/_fine_cal.py [hunk 1]
```diff
@@ -388,4 +390,14 @@ def write_fine_calibration(fname, calibration):
     """
     _check_fname(fname, overwrite=True)
     check_fname(fname, 'cal', ('.dat',))
-    raise NotImplementedError
+
+    # TODO: Figure out if write or write+binary mode is best
```
Comment (Eric89GXL) [inline mne/preprocessing/_fine_cal.py]: use `wb` and e.g. `cal_file.write((cal_str + '\n').encode('ASCII'))`, it's clear you're writing an ASCII-encoded file that way

**Comment by wronk** [comment]
>I would probably make both deal with the "complete representation", i.e. the dict. Then whatever uses read_fine_cal should take the dict convert to cal_chs and cal_ch_numbers. Make sense?

Yep, sounds good. The inconsistency is actually in `maxwell.py` -- sorry for the mistake. I added a checkbox to update that function as soon as this gets merged.

**Comment by wronk** [comment]
@Eric89GXL, ready for review :)

**Comment by Eric89GXL** [comment]
Nice, thanks @wronk

**Pull request merged**